fern/openapi-ir/definition/example.yml

```diff
@@ -0,0 +1,112 @@
+types:
+  SchemaInstanceId:
+    docs: |
+      Uniquely identifies an instance of a schema. For nested instances,
+      the id will be a dot-delimited path, e.g. foo.bar.baz.
+    type: string
+
+  Example:
+    docs: |
+      An OpenAPI example, which is either fully specified, partially
+      specified, or a reference to another schema instance's example.
+    union:
+      full: FullExample
+      partial: PartialExample
+      reference: ReferencedExample
+
+  FullOrPartialExample:
+    docs: |
+      Specifically designed for the PartialObjectExample.
+      The included properties can either be full or partial
+      examples, but not references.
+    union:
+      full: FullExample
+      partial: PartialExample
+
+  FullExample:
+    union:
+      primitive: PrimitiveExample
+      object: FullObjectExample
+      array: ArrayExample
+      map: MapExample
+      optional: FullExample
+      enum: EnumExample
+      reference: FullExample
+      literal: LiteralExample
+      oneOf: FullOneOfExample
+      nullable: FullExample
+      unknown: {}
+
+  PrimitiveExample:
+    union:
+      int: integer
+      int64: long
+      float: double
+      double: double
+      string: string
+      datetime: datetime
+      date: date
+      base64: base64
+      boolean: boolean
+
+  FullObjectExample:
+    properties:
+      properties: map<string, FullExample>
+
+  ArrayExample:
+    properties:
+      elements: list<FullExample>
+
+  MapExample:
+    properties:
+      pairs: list<MapExamplePair>
+
+  MapExamplePair:
+    properties:
+      key: PrimitiveExample
+      value: FullExample
+
+  EnumExample:
+    properties:
+      examples: list<EnumValueExample>
+
+  EnumValueExample:
+    properties:
+      value: string
+
+  FullOneOfExample:
+    union:
+      discriminated: FullDiscriminatedOneOfExample
+      undisciminated: FullUnDiscriminatedOneOfExample
+
+  FullDiscriminatedOneOfExample:
+    properties:
+      examples: map<string, FullExample>
+
+  FullUnDiscriminatedOneOfExample:
+    properties:
+      examples: list<FullExample>
+
+  ReferencedExample:
+    properties:
+      reference: SchemaInstanceId
+
+  PartialExample:
+    union:
+      object: PartialObjectExample
+      oneOf: PartialOneOfExample
+
+  PartialObjectExample:
+    properties:
+      includedProperties: map<string, FullOrPartialExample>
```

Comment:
@dsinghvi heads up also made a small adjustment to `includedProperties` so that it can be either a `FullExample` or a `PartialExample` (but not a reference because I _think_ that's impossible in the example syntax).

Reply:
looks great as is!